Solutions.txt

```diff
@@ -0,0 +1,1410 @@
+| 2007 | Pereira | Rome   |    100 | 1004 |
++------+---------+--------+--------+------+
+
+
+3) Find the largest order taken by each salesperson on each date. 
+
+SELECT MAX(amt) max_amount, snum, odate FROM orders GROUP BY odate, snum;
++------------+------+------------+
+| max_amount | snum | odate      |
++------------+------+------------+
+|    1098.16 | 1007 | 1990-10-03 |
+|     767.19 | 1001 | 1990-10-03 |
+|    1900.10 | 1004 | 1990-10-03 |
+|    5160.45 | 1002 | 1990-10-03 |
+|    1713.23 | 1003 | 1990-10-04 |
+|      75.75 | 1002 | 1990-10-04 |
+|    4723.00 | 1001 | 1990-10-05 |
+|    1309.95 | 1002 | 1990-10-06 |
+|    9891.88 | 1001 | 1990-10-06 |
++------------+------+------------+
+
+4) Arrange the Order table by descending customer number. 
+
+SELECT * FROM orders ORDER BY cnum DESC;
++------+---------+------------+------+------+
+| onum | amt     | odate      | cnum | snum |
++------+---------+------------+------+------+
+| 3001 |   18.69 | 1990-10-03 | 2008 | 1007 |
+| 3006 | 1098.16 | 1990-10-03 | 2008 | 1007 |
+| 3002 | 1900.10 | 1990-10-03 | 2007 | 1004 |
+| 3008 | 4723.00 | 1990-10-05 | 2006 | 1001 |
+| 3011 | 9891.88 | 1990-10-06 | 2006 | 1001 |
+| 3007 |   75.75 | 1990-10-04 | 2004 | 1002 |
+| 3010 | 1309.95 | 1990-10-06 | 2004 | 1002 |
+| 3005 | 5160.45 | 1990-10-03 | 2003 | 1002 |
+| 3009 | 1713.23 | 1990-10-04 | 2002 | 1003 |
+| 3003 |  767.19 | 1990-10-03 | 2001 | 1001 |
++------+---------+------------+------+------+
+
+
+5) Find which salespeople currently have orders in the order table. 
+
+SELECT DISTINCT s.* FROM orders o, salespeople s where o.snum=s.snum;
++------+---------+-----------+------+
+| snum | sname   | city      | comm |
++------+---------+-----------+------+
+| 1007 | Rifkin  | Barcelona |   15 |
+| 1001 | Peel    | London    |   12 |
+| 1004 | Motika  | London    |   11 |
+| 1002 | Serres  | SanJose   |   13 |
+| 1003 | AxelRod | New York  |   10 |
++------+---------+-----------+------+
+
+
+6) List names of all customers matched with the salespeople serving them. 
+
+SELECT c.cname, s.sname FROM customers c, salespeople s WHERE s.snum=c.snum;
++----------+---------+
+| cname    | sname   |
++----------+---------+
+| Hoffman  | Peel    |
+| Giovanni | AxelRod |
+| Liu      | Serres  |
+| Grass    | Serres  |
+| Clemens  | Peel    |
+| Cisneros | Rifkin  |
+| Pereira  | Motika  |
++----------+---------+
+
+
+7) Find the names and numbers of all salespeople who have more than one customer. 
+
+SELECT c.snum, s.sname, count(cnum) 'number of customers' FROM customers c, salespeople s WHERE s.snum=c.snum Group By s.snum Having COUNT(cnum)>1;
++------+--------+---------------------+
+| snum | sname  | number of customers |
++------+--------+---------------------+
+| 1001 | Peel   |                   2 |
+| 1002 | Serres |                   2 |
++------+--------+---------------------+
+
+
+8) Count the orders of each of the salespeople and output the results in descending order. 
```

Comment:
Wrong Edge case missed here
Try to think about this
if there is salespeople who is not associated with any order
Expected
In that case the order count of salespeople shows as 0.